### PR TITLE
Added subsection support to the FlowsheetDataGrid

### PR DESCRIPTION
**Pull request by cm9361**
### Summary

**What was changed:**
Subsection support was added to the FlowsheetDataGrid component.

**Why it was changed:**
The change was made to satisfy a functional requirement of a consumer of the FlowsheetDataGrid component.

### Testing

This change was tested using:

- [X] WDIO
- [X] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [X] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-10360 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-framework-docs/src/terra-dev-site/doc/data-grid/FlowsheetDataGrid.4/About.5.doc.mdx [hunk 1]
```diff
@@ -136,6 +136,33 @@ A column specifies the data to render a cell in the header row of the Flowsheet
...
+| **isCollapsible** | Bool   | Optional                                        | False         | A Boolean value indicating whether the the section is collapsible.                                                  |
+| **isCollapsed**   | Bool   | Optional                                        | False         | A Boolean value indicating whether the section is collapsed. If true, the contents of the section is not displayed. |
+| **text**          | String | Optional                                        | None          | A text string to render in the section header.                                                                      |
+| **rows**          | Array  | Optional                                        | []            | An array of row objects to be rendered in the section.                                                              |
+| **subsections**   | Array  | Optional                                        | []            | An array of subsections objects to be rendered within the section.                                                  |
+
+#### Subsection
```
Comment (adoroshk) [inline packages/terra-framework-docs/src/terra-dev-site/doc/data-grid/FlowsheetDataGrid.4/About.5.doc.mdx:155]: Shouldn't it be the same level header like Section: `### Subsection`? Looks smaller than Section and Rows headers.

**Review by adoroshk — APPROVED** [review]

File: packages/terra-framework-docs/src/terra-dev-site/doc/data-grid/FlowsheetDataGrid.4/About.5.doc.mdx (hunk 1 above)
Comment (cm9361) [reply packages/terra-framework-docs/src/terra-dev-site/doc/data-grid/FlowsheetDataGrid.4/About.5.doc.mdx:155]: @adoroshk - This is a good call out.  I copied the structure from the Terra Table component where it originates.  However, I could update this style for both components.  I do think that is appropriate.

**Comment by mjpalazzo** [comment]
@cm9361 - I validated with Chrome, Edge, and JAWS with Edge. The KB navigation works as required. Screen reader navigation and announcements work as required. I observed one minor issue. If I am on a row in the body of the grid and press the End key (fn+right arrow on my MacBook), JAWS announces the labels of all the sections and subsections, regardless where the focus is located. For example, if focus is in column 6 row 4 in section 1, subsection 1, JAWS announces "End, Primary Contact, Test Section, Test Subsection, Test Subsection number 2, Test Section number 2, Quinzel, Harleen, Column 6"
Some thing similar happens if I navigate from a column header with the Down Arrow. For example when I navigate the last column to column 6 row 4. JAWS announces, "Fleck, Arthur, Primary Contact, Test Section, Test Subsection, Test Subsection number 2, Test Section number 2, Quinzel, Harleen, Column 6 row 4"
This happens for columns 2 - 5 for any cell that follows a section or subsection header.

**Comment by cm9361** [comment]
> @cm9361 - I validated with Chrome, Edge, and JAWS with Edge. The KB navigation works as required. Screen reader navigation and announcements work as required. I observed one minor issue. If I am on a row in the body of the grid and press the End key (fn+right arrow on my MacBook), JAWS announces the labels of all the sections and subsections, regardless where the focus is located. For example, if focus is in column 6 row 4 in section 1, subsection 1, JAWS announces "End, Primary Contact, Test Section, Test Subsection, Test Subsection number 2, Test Section number 2, Quinzel, Harleen, Column 6" Some thing similar happens if I navigate from a column header with the Down Arrow. For example when I navigate the last column to column 6 row 4. JAWS announces, "Fleck, Arthur, Primary Contact, Test Section, Test Subsection, Test Subsection number 2, Test Section number 2, Quinzel, Harleen, Column 6 row 4" This happens for columns 2 - 5 for any cell that follows a section or subsection header.

@mjpalazzo - Thank you for the feedback.  Isn't the issue you described a known pre-existing issue?  I thought we landed on the fact that the attributes were semantically correct in this scenario.  If it is the pre-existing issue and we want to address it somehow, do we need a JIRA for the work.  I thought we had one open previously.

**Comment by mjpalazzo** [comment]
> > @cm9361 - I validated with Chrome, Edge, and JAWS with Edge. The KB navigation works as required. Screen reader navigation and announcements work as required. I observed one minor issue. If I am on a row in the body of the grid and press the End key (fn+right arrow on my MacBook), JAWS announces the labels of all the sections and subsections, regardless where the focus is located. For example, if focus is in column 6 row 4 in section 1, subsection 1, JAWS announces "End, Primary Contact, Test Section, Test Subsection, Test Subsection number 2, Test Section number 2, Quinzel, Harleen, Column 6" Some thing similar happens if I navigate from a column header with the Down Arrow. For example when I navigate the last column to column 6 row 4. JAWS announces, "Fleck, Arthur, Primary Contact, Test Section, Test Subsection, Test Subsection number 2, Test Section number 2, Quinzel, Harleen, Column 6 row 4" This happens for columns 2 - 5 for any cell that follows a section or subsection header.
> 
> @mjpalazzo - Thank you for the feedback. Isn't the issue you described a known pre-existing issue? I thought we landed on the fact that the attributes were semantically correct in this scenario. If it is the pre-existing issue and we want to address it somehow, do we need a JIRA for the work. I thought we had one open previously.

@cm9361 - you are correct. I had forgotten that this is a pre-existing issue. We may already have a Jira for it. I will follow up with Chris. As I said it is minor and does not affect funtionality, so we will deal with it in the future. Your changes are working well so I approve.

**Comment by chrismichalewicz** [comment]
@mjpalazzo @eawww When navigating to the Grid it announces the # of rows and columns, should it also announce the number of sections contained within the grid?   I was unable to find an ARIA example that had sections to test to see how they handled it.

**Comment by cm9361** [comment]
> @mjpalazzo @eawww When navigating to the Grid it announces the # of rows and columns, should it also announce the number of sections contained within the grid? I was unable to find an ARIA example that had sections to test to see how they handled it.

@chrismichalewicz - The table does not have a concept of sections.  That is functionality that we added.  The screen reader would not have a way to announce that.

**Comment by chrismichalewicz** [comment]
> > @mjpalazzo @eawww When navigating to the Grid it announces the # of rows and columns, should it also announce the number of sections contained within the grid? I was unable to find an ARIA example that had sections to test to see how they handled it.
> 
> @chrismichalewicz - The table does not have a concept of sections. That is functionality that we added. The screen reader would not have a way to announce that.

Thanks @cm9361 for clarifying! It works well for me as well, great job! +1 I will also add the Functional Verified label.

**Comment by mjpalazzo** [comment]
> > > @mjpalazzo @eawww When navigating to the Grid it announces the # of rows and columns, should it also announce the number of sections contained within the grid? I was unable to find an ARIA example that had sections to test to see how they handled it.
> > 
> > 
> > @chrismichalewicz - The table does not have a concept of sections. That is functionality that we added. The screen reader would not have a way to announce that.
> 
> Thanks @cm9361 for clarifying! It works well for me as well, great job! +1 I will also add the Functionally Reviewed label.

Agreed...WAI and MDN have notthing to say about tables with sections.

**Pull request merged**